test/e2e/upgrade_clusterctl_test.go

```diff
@@ -62,6 +63,7 @@ var _ = Describe("When testing cluster upgrade from releases (v1.9=>current) [cl
 			InitWithBootstrapProviders:      []string{fmt.Sprintf(providerKubeadmPrefix, capiStableRelease)},
 			InitWithControlPlaneProviders:   []string{fmt.Sprintf(providerKubeadmPrefix, capiStableRelease)},
 			InitWithInfrastructureProviders: []string{fmt.Sprintf(providerMetal3Prefix, capm3StableRelease)},
+			InitWithIPAMProviders:           []string{fmt.Sprintf(providerMetal3ipamPrefix, capm3StableRelease)},
```

Comment:
This could lead to issues because the versions of our CAPM3 and IPAM releases are not always synchronized. For example, the latest IPAM release is currently 1.9.4, while CAPM3 is at version 1.9.2.
you need to change GetCAPM3StableReleaseOfMinor function, perhaps make it more generic so it can be used for both capm3 and ipam.

Reply:
I agree, we must not assume anything about the versions between IPAM and CAPM3.